corelog/README.md

```diff
@@ -0,0 +1,43 @@
+# CoreOS Log
+
+There are far too many logging packages out there, with varying degrees of licenses, far too many features (colorization, all sorts of log frameworks) or are just a pain to use (lack of `Fatalln()`?)
+
+## Design Principles
+
+* Logging is off by default, except critical failures, which are unrecoverable.
+
+Log functions are helpful but silenced until explicitly activated.
+
+* `package main` is the place where logging gets turned on and routed
+
+A library should not touch log options, only generate log entries. Libraries are silent until main lets them speak.
+
+* All log options are runtime-configurable. 
+
+Still the job of `main` to expose these configurations. `main` may delegate this to, say, a configuration webhook, but does so explicitly. 
+
+* There is one log object per package. It is registered under its repository and package name.
+
+`main` activates logging for its repository and any dependency repositories it would also like to have output in its logstream. `main` also dictates at which level each subpackage logs.
+
```

Comment:
So main needs to know all repos the project uses? What if main just imports "github.com/foo/..." and it in turn imports "github.com/bar/..." -- is main expected to know that it needs to turn on logging for "github.com/bar"?